File: 170307_ActivityInClass/app/src/main/java/com/mobiledevices/miguel/a170307_activityinclass/JSONAdapter.java
package com.mobiledevices.miguel.a170307_activityinclass;

import android.app.Activity;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * Created by miguel on 7/03/17.
 */

public class JSONAdapter extends BaseAdapter {

    private JSONArray data;
    private Activity activity;

    public JSONAdapter(JSONArray d, Activity a) {

        this.data = d;
        this.activity = a;
    }

    @Override
    public int getCount() {
        return data.length();
    }

    @Override
    public Object getItem(int position) {

        try {

            return data.getJSONObject(position);
        }
        catch(JSONException jse) {

            jse.printStackTrace();
        }

        return null;
    }

    @Override
    public long getItemId(int position) {

        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {

        if(convertView == null) {

            convertView = this.activity.getLayoutInflater().inflate(R.layout.row, null);
        }

        TextView name = (TextView) convertView.findViewById(R.id.Large);
        TextView gradeText = (TextView) convertView.findViewById(R.id.Medium);

        try {
            JSONObject current = this.data.getJSONObject(position);
            name.setText(current.getString("name"));
            gradeText.setText(current.getDouble("grade") + "");
        }
        catch(JSONException jse) {
            jse.printStackTrace();
        }

        return null;
    }
}

File: Activity3.3/Activity 3.3/JSONAdapter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Java.Lang;

namespace Activity_3._3
{
    class JSONAdapter : BaseAdapter
    {
        Activity activity;
        JavaList<Post> data;

        public JSONAdapter(Activity activity, JavaList<Post> data)
        {
            this.activity = activity;
            this.data = data;
        }

        public override int Count => data.Count;

        public override Java.Lang.Object GetItem(int position)
        {
            return data[position];
        }

        public override long GetItemId(int position)
        {
            return data[position].id;
        }

        public override View GetView(int position, View convertView, ViewGroup parent)
        {
            if (convertView == null)
            {
                convertView = LayoutInflater.From(this.activity).Inflate(Resource.Layout.JSON_Row, null);
            }

            TextView rowTitle = this.activity.FindViewById<TextView>(Resource.Id.tv_row_title);
            TextView rowBody = this.activity.FindViewById<TextView>(Resource.Id.tv_row_body);
            Post post = (Post)this.GetItem(position);

            rowTitle.Text = post.title;
            rowBody.Text = post.body;

            return convertView;
        }
    }
}